ReactCommon/hermes/inspector/chrome/Connection.cpp

```diff
@@ -1411,20 +1411,14 @@ Connection::Impl::makePropsFromValue(
 }
 
 void Connection::Impl::handle(const m::runtime::GetHeapUsageRequest &req) {
-  auto resp = std::make_shared<m::runtime::GetHeapUsageResponse>();
-  resp->id = req.id;
-
-  inspector_
-      ->executeIfEnabled(
```

Comment:
executeIfEnabled() ensures that the JS VM is not running when making calls to it -- it therefore serializes accesses to the VM. I am not sure it is safe to make this change. @neildhar, any thoughts?

Reply:
I think in this case we want it to run if the js vm is running since this command gets called periodically by devtools to get the current memory usage.

Reply:
I had a quick look and it seems fine to call from any thread since this info comes from the GC and uses a lock.

Reply:
getHeapUsage() is only synchronized with the GC, not with the VM execution. Therefore it is not safe to be invoked with the VM running. You could use onPause/onResume to determine whether to invoke getHeapUsage(), or whether return a cached value.

Reply:
I think it would be nice to be able to run getHeapUsage while the VM is running, it is used to display realtime memory usage in chrome devtools, so requiring that the VM is paused kind of defeats the purpose of this.

<img width="368" alt="image" src="https://user-images.githubusercontent.com/2677334/177457702-d77386eb-9a17-4443-b639-ceee62a5b60c.png">

Is there any info coming from the VM that needs synchronization? Also can anything bad happen if we execute this on a running VM, except from potential wrong values, which I think is fine in this case.

Reply:
I could also just disable this for now if we're not comfortable landing this change. This feature is not very important compared to be able to profile.